### PR TITLE
BTT SKR V1.1, V1.3, V1.4, V1.4 Turbo - Added missing pins for E1 stepper and added definition of E1 as Z2 Stepper

### PR DESCRIPTION
**Pull request by dan-and**
### Requirements

* It doesn't change anything for users who don't use E1 nor Z2, since the definitions are added. 

### Description

I saw some confusion of how to use the SKR V1.3 (and the same applies to V1.1 and V1.4/V1.4 Turbo) in a Dual Z-stepper setup. 

I investigated the src/pins/lpc176x/pins_BTT_SKR_V1_x.h definitions and figured out that the pin assignments for E1 stepper were just partial done or completly missing. 

I also added the definition of Z2 on the E1 pins, so the user can utilize the E1 Stepper for Dual-Z motors just by changing the corresponding Configuration_adv.h and not fiddling around with the SKR pin definitions. 

### Benefits

a) The user can finally use the E1 stepper without doing manual changes to the SKR pin descriptions. 
b) The user can utilize the E1 Stepper for Dual-Z motors just by changing the corresponding Configuration_adv.h and not fiddling around with the SKR pin definitions.

### Configurations

Basically just add a stepper driver of your choice on E1 and either add
#define E1_DRIVER_TYPE  
as usual 

or if you want to test dual z stepper setups: 
Configuration.h 
#define Z2_DRIVER_TYPE 
and 
Configuration_adv.h 
#define NUM_Z_STEPPER_DRIVERS 2 

to test Dual-Z stepper motor setups on BigTreeTech V1.1, V1.3, V1.4 and V1.4 Turbo

### CODE REVIEW

File: Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h [hunk 1]
```diff
@@ -52,6 +52,14 @@
 #define E0_DIR_PIN                         P2_13
 #define E0_ENABLE_PIN                      P2_12
 
+#define E1_STEP_PIN                        P0_01
+#define E1_DIR_PIN                         P0_00
+#define E1_ENABLE_PIN                      P0_10
+
+#define Z2_STEP_PIN                        P0_01 // Reusing E1 Stepper Port as Z2 for Dual Z-Stepper
+#define Z2_DIR_PIN                         P0_00 // E1 as Z2
+#define Z2_ENABLE_PIN                      P0_10 // E1 as Z2
```
Comment (sjasonsmith) [inline Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h:61]: Z2 will auto-reassign onto an unused E1 if available. You don't need to call any Z2 stuff out in the pins files.

File: Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h [hunk 2]
```diff
@@ -52,6 +52,14 @@
 #define E0_DIR_PIN                         P2_13
 #define E0_ENABLE_PIN                      P2_12
 
+#define E1_STEP_PIN                        P0_01
+#define E1_DIR_PIN                         P0_00
+#define E1_ENABLE_PIN                      P0_10
```
Comment (sjasonsmith) [inline Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h:57]: The E1 pins are already defined in `pins_BTT_SKR_common.h`. You shouldn't need to redefine them for any of these SKR boards.

**Review by sjasonsmith — COMMENTED** [review]
I don't think these changes should have been necessary. If it wasn't working without them there might be some minor bug, but this should have been handled by the common header, and automatic reassignment of Z2 drivers.

File: Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h (hunk 1 above)
Comment (thisiskeithb) [reply Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h:61]: Can confirm that dual Z remapping is automatic on these boards (and for Marlin in general) when you’re using a single E driver.

File: Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h (hunk 2 above)
Comment (dan-and) [reply Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h:57]: you are right. Sorry for the pull request.

File: Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h (hunk 1 above)
Comment (dan-and) [reply Marlin/src/pins/lpc1768/pins_BTT_SKR_V1_1.h:61]: you are right. Sorry for the pull request.
